primary.c

```diff
@@ -1,30 +1,56 @@
-#include<stdio.h>
-#include<math.h>
-#include<stdbool.h>
-#include<string.h>
+#include <stdio.h>
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
 #include <stdlib.h>
-
-bool isPrime(int number){
-	for (int i = 2; i <= sqrt(number); i++)
-	{
-		if (number % i == 0)
-			return false;
-	}
-	return true;
-}
+#include <sys/types.h>
+#include <unistd.h>
+#include "defs.h"
 
 int main(int argc, char *argv[]){
-	if(argc == 1){
-		printf("You don't type any number, Please type numberes throw argv");
-		return 0;
-	}
-	int num;
-	int i = 0;
-	for(i = 1; i < argc; i++){
-		num = atoi(argv[i]);
-		if(isPrime(num)){
-			printf("%d ", num);
+    int pipe_fd[2], c, status;
+    pid_t p;
+    FILE *f;
+
+    if (pipe(pipe_fd) == -1){
+            fprintf(stderr, "Pipe Failed");
+			CLEAN(EXIT_FAILURE);
+    }
+
+    p = fork();
+    switch(p) {
+	case -1:
+		fprintf(stderr, "Fork Failed");
+		CLEAN(EXIT_FAILURE);
+		break;
+	case 0:
+		if (argc == 1){
+			printf("You don't type any number, Please type numberes through argv");
+			CLEAN(EXIT_FAILURE);
+		}
+		dup2(pipe_fd[1], 1);
+		close(pipe_fd[0]);
+		execve("./is_real_prime", argv, NULL);
+		break;
+	default:
+		close(pipe_fd[1]);
+		f = fdopen(pipe_fd[0], "r");
+		c = getc(f);
+		while (c != EOF){
+			putchar(c);
+			c = getc(f);
+			if (c == EOF)
+				break;
+			printf(", ");
 		}
+		CLEAN(EXIT_SUCCESS);
+		exit(0);
+	}
+cleanup:
+	if(pipe(pipe_fd) == 0)
```

Comment:
Why do you run pipe again?
You need to check if one of the fds is set, and if so - close it.